torch/csrc/autograd/engine.cpp

```diff
@@ -661,48 +725,52 @@ void Engine::enqueue_blocked_task_on_cpu(NodeTask task) {
       std::move(task), /* incrementOutstandingTasks */ false);
 }
 
+bool graph_task_completed(const GraphTask& graph_task) {
+  return graph_task.outstanding_tasks_.load() == 0 ||
+      (graph_task.exit_on_error_ && graph_task.has_error_.load());
```

Comment:
Now I am wondering... when should we ever NOT exit on error?

Reply:
Well, I didn't want to change the behavior of the current autograd engine and hence I added this logic. I thought there was a good reason we continue executing other tasks even though we hit an error for one.

Reply:
TBH, I think it's an oversight in the current engine. @albanD, @apaszke any ideas?